### PR TITLE
[MXNET-908] Enable minimal OSX Travis build

### PR DESCRIPTION
**Pull request by KellenSunderland**
## Description ##
This PR removes some legacy Travis code, and updates Travis to run cached MXNet builds on the platform.  It is the first step in re-enabling MacOS testing.  Subsequent PRs will enable testing (which will rely on builds first being cached).  Subsequent PRs can also re-enable some of the features removed in this PR, for instance possible running more tests in addition to python tests, and re-enabling the model-download caching folder.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,108 +1,28 @@
...
-
-notifications:
-# Emails are sent to the committer's git-configured email address by default,
-  email:
-    on_success: change
-    on_failure: always
-  #slack: dmlc:NmroCzntCiWOuxUZpii40USd
+  - export MXNET_STORAGE_FALLBACK_LOG_VERBOSE=0
+  - mv make/osx.mk config.mk
+  - make -j 2
```
Comment (marcoabreu) [inline .travis.yml]: Is the variable not available in Travis?

Comment (KellenSunderland) [reply .travis.yml]: Referring to the magic '2' here, good point, I don't specify why that's required.  I've updated with a comment that should provide some background.

**Comment by KellenSunderland** [comment]
Example build for this commit: https://travis-ci.org/KellenSunderland/incubator-mxnet/builds/424768656

**Comment by KellenSunderland** [comment]
@mxnet-label-bot can please add [pr-awaiting-review]

**Pull request merged**